wiki/DEFINITION_OF_DONE_CRITERIA.md

```diff
@@ -0,0 +1,99 @@
+- Public API contains a C-Wrapper around the stateless API
+- Public API contains an Scikit-learn-like stateful wrapper around the stateless API
+
+#### Testing
+
+- Prims: GTests with different inputs
+- Algorithms: End-to-end GTests with different inputs and different datasets
+
+#### Documentation
+
+- Complete and comprehensive Doxygen strings explaining the public API, restrictions, and gotchas. Any array parameters should also note whether the underlying memory is host or device. 
+
+
+### Python
+
+#### Design
+
+- Python class is as "near drop-in replacement" for Scikit-learn (or relevant industry standard) API as possible
+    - It is recommended to open an initial PR with the API design if there are going to be significant differences with reference APIs, or lack of a reference API, to have a discussion about it. 
+- Python class is pickleable
+- APIs use `input_to_dev_array` to accept flexible inputs and check their datatypes
+
+#### Testing 
+
+- Pytests for wrapper functionality against Scikit-learn using relevant datasets
+- Stress tests against reasonable inputs (e.g short-wide, tall-narrow, different numerical precision)
+- Pytests for pickle capability
+- Pytests to evaluate correctness against Scikit-learn on a variety of datasets
+- Add algorithm to benchmarks packages
+
+#### Documentation
+
+- Complete and comprehensive Pydoc strings explaining public API, restrictions, and gotchas
+- Docstrings include references to the underlying algorithm (e.g paper or Scikit-learn algorithm being implemented or a description of the algorithm used if nonstandard).
+
+
+## Algorithm Status
+
+Sometimes it is possible that only some of these requirements are satisfied before an algorithm gets released. We assign a status of `bronze`, `silver`, or `gold` to denote the amount of work necessary to provide the most robust, performant, and maintainable algorithm possible.
+
+Note that a change to an algorithm's classification can go up or down, depending on circumstances. For example, an algorithm that has been considered `Gold status` might drop to a `Silver status` when being updated to a new low-level feature or library. 
+
+### Bronze: 
+- Known design flaws affecting sustainability and/or performance require a significant effort / redesign / refactor
+- C++ checklist / API functionality is complete but Python API has not been built
+- ure Python API exists without using C++ for processing when a C++ implementation would provide a significant performance boost
+- There are known issues that affect memory / accuracy / stability on some inputs
+- Coverage of automated tests in C++ or Python is minimal and could be significantly improved
+- Outdated / legacy code that requires significant work and refactoring
+
+
+### Silver:
+- Known design flaws affecting sustainability and/or performance are fairly low in complexity (can be fixed quickly and without too much effort)
```

Comment:
I personally think we should focus the levels on end-user and developer experience rather than effort needed to improve. So, a design flaw affecting stability that is very minor but requires a *huge* effort to solve could still exist in silver...